backend/src/packages/groups/group.service.ts

```diff
@@ -0,0 +1,55 @@
+import { type IService } from '~/libs/interfaces/interfaces.js';
+import { type GroupRepository } from '~/packages/groups/group.repository.js';
+
+import { GroupEntity } from './group.entity.js';
+import {
+  type GroupDatabaseModelCreateUpdate,
+  type GroupEntityObjectT,
+  type GroupEntityT,
+} from './libs/types/types.js';
+
+class GroupService implements IService<GroupEntityObjectT> {
+  private groupRepository: GroupRepository;
+
+  public constructor(groupRepository: GroupRepository) {
+    this.groupRepository = groupRepository;
+  }
+
+  public async findByName(name: GroupEntityT['name']): Promise<GroupEntityObjectT | null> {
+    const result = await this.groupRepository.find({ name });
+
+    return result.length === 1 ? GroupEntity.initialize(result[0]).toObject() : null;
+  }
```

Comment:
I believe we shouldn't look up for a group by it's name, only by key.
Also avoid magic numbers, prefer destructuring

```const [group = null] = await ;
return group ? GroupEntity...
```